pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/AsciidoctorService.java

```diff
@@ -176,10 +177,10 @@ private String buildDocument(@Nonnull Document base, @Nonnull Locale locale, @No
                     + ",variant: " + variantName + ", draft: " + isDraft);
         }
 
-        // Use a service-level resource resolver to build the module as it will require write access to the resources
+        // Use a service-level resource resolver to build the module or assemblies as it will require write access to the resources
         try (ResourceResolver serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver()) {
 
-            Class cls = base instanceof Module ? Module.class : Assembly.class;
+            Class cls = base.getResourceType().equals(PantheonConstants.RESOURCETYPE_ASSEMBLY) ? Assembly.class : Module.class;
             Document serviceDocument = (Document) SlingModels.getModel(serviceResourceResolver, base.getPath(), cls);
```

Comment:
This is interesting. It looks like we're putting effort into figuring out whether the resource is a module or an assembly, but then we're ultimately discarding the answer and just casting to a generic document class for the rest of the method. I bet this could be removed.

I know you didn't write this originally, but since it looks like you have a conflict to resolve anyway, bonus points for taking care of this at the same time. :)

Reply:
@benradey  I have created a separate ticket, to track this code quality issue, we can work on it separately.
https://projects.engineering.redhat.com/browse/CCS-3772